apps/accounts/views.py

```diff
@@ -38,3 +52,25 @@ class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
 
     def retrieve(self, request, *args, **kw):
         return Response(self.get_serializer(request.user).data)
+
+
+class RegisterView(APIView):
+
+    @waffle_switch('fxa-auth')
+    def post(self, request):
+        if 'code' not in request.DATA:
+            return Response({'error': 'No code provided.'}, status=400)
+
+        try:
+            identity = verify.fxa_identify(request.DATA['code'],
+                                           config=settings.FXA_CONFIG)
+        except verify.IdentificationError:
+            return Response({'error': 'Profile not found.'}, status=401)
+        if find_user(identity):
+            return Response({'error': 'That account already exists.'},
+                            status=400)
+        else:
+            user = UserProfile.objects.create_user(
+                email=identity['email'], username=None, fxa_id=identity['uid'])
```

Comment:
Just to check UserProfile has to have a unique username, but MySQL ignores `None` in the uniqueness so this is ok?